src/test/java/com/googleinterns/zoomtube/servlets/AuthenticationServletTest.java

```diff
@@ -0,0 +1,140 @@
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.googleinterns.zoomtube.data.AuthenticationStatus;
+import com.ryanharter.auto.value.gson.GenerateTypeAdapter;
+
+public class AuthenticationServletTest {
+  private AuthenticationServlet servlet;
+  private MockHttpServletRequest request;
+  private MockHttpServletResponse response;
+
+  @Before
+  public void setUp() throws ServletException {
+    servlet = new AuthenticationServlet();
+    servlet.init();
+    request = new MockHttpServletRequest();
+    response = new MockHttpServletResponse();
+  }
+
+  @Test
+  public void doGet_loggedIn_true() throws ServletException, IOException {
+    LocalServiceTestHelper helper =
+        new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("example.com")
+          .setEnvEmail("test@example.com");
+    helper.setUp();
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    String json = response.getContentAsString();
+    Gson gson = new GsonBuilder()
+        .registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY)
+        .create();
+    AuthenticationStatus status = gson.fromJson(json, AuthenticationStatus.class);
+    assertThat(status.loggedIn()).isTrue();
+
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGet_loggedIn_false() throws ServletException, IOException {
+    LocalServiceTestHelper helper =
+        new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+          .setEnvIsLoggedIn(false);
+    helper.setUp();
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    String json = response.getContentAsString();
+    Gson gson = new GsonBuilder()
+        .registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY)
+        .create();
+    AuthenticationStatus status = gson.fromJson(json, AuthenticationStatus.class);
+    assertThat(status.loggedIn()).isFalse();
+
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGet_user_email() throws ServletException, IOException {
+    final String EMAIL = "test@example.com";
+    LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+        .setEnvIsLoggedIn(true)
+        .setEnvAuthDomain("example.com")
+        .setEnvEmail(EMAIL);
+    helper.setUp();
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    String json = response.getContentAsString();
+    Gson gson = new GsonBuilder()
+        .registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY)
+        .create();
+    AuthenticationStatus status = gson.fromJson(json, AuthenticationStatus.class);
+    assertThat(status.user().get().getEmail()).isEqualTo(EMAIL);
+
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGet_loggedInUrls() throws ServletException, IOException {
```

Comment:
nit: Does it return more than 1 url? Should Urls be plural?

Reply:
I check that it returned the logoutUrl but not the loginUrl, so it is checking two url fields

Reply:
should I rename these last two to doGet_urls_whenLoggedIn and doGet_urls_whenLoggeOut?